### ISSUE TITLE
IntelliCode Completions does not work at all on VSCode >= v1.82 due to updated node

### ISSUE THREAD

**Issue by leet0rz** [MicrosoftDocs/intellicode#491]
Type: <b>Bug</b>

Been using vscode as usual and just updated to the most recent vscode while using intellicode.

Extension version: 1.0.22
VS Code version: Code 1.82.0 (8b617bd08fd9e3fc94d14adb8d358b56e3f72314, 2023-09-06T22:07:07.438Z)
OS version: Windows_NT x64 10.0.19044
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 3600 6-Core Processor               (12 x 3593)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.93GB (7.83GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by xGanimeDx** [comment]
faced same issue

**Comment by CleyFaye** [comment]
I can confirm with some logs.

Versions:

- VSCode: 1.82.2
- Intellicode extension: 1.2.30
- Intellicode Completions extension: 1.0.22

VSCode about:
Version: 1.82.2
Commit: abd2f3db4bdb28f9e95536dfa84d8479f1eb312d
Date: 2023-09-14T05:51:20.981Z
Electron: 25.8.1
ElectronBuildId: 23779380
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Linux x64 5.15.0-83-generic

Log output for "Intellicode Completion":

```raw
Extension version: 1.0.22 034a5c7
[Worker thread] Unhandled rejection error: Failed to parse URL from /home/cleyfaye/.vscode/extensions/visualstudioexptteam.vscodeintellicode-completions-1.0.22/dist/tree-sitter.wasm
[Worker thread] Unhandled rejection error stack: TypeError: Failed to parse URL from /home/cleyfaye/.vscode/extensions/visualstudioexptteam.vscodeintellicode-completions-1.0.22/dist/tree-sitter.wasm
    at Object.fetch (node:internal/deps/undici/undici:11413:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Worker thread] Unhandled rejection error: Failed to parse URL from /home/cleyfaye/.vscode/extensions/visualstudioexptteam.vscodeintellicode-completions-1.0.22/dist/blingfire.wasm
[Worker thread] Unhandled rejection error stack: TypeError: Failed to parse URL from /home/cleyfaye/.vscode/extensions/visualstudioexptteam.vscodeintellicode-completions-1.0.22/dist/blingfire.wasm
    at Object.fetch (node:internal/deps/undici/undici:11413:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[inlineCompletionProvider.provideInlineCompletionItems() 1]  Caught error when requesting inline completion: Error: Operation took longer than 5000ms.
[inlineCompletionProvider.provideInlineCompletionItems() 1] Unexpected exception: Operation took longer than 5000ms.
Error: Operation took longer than 5000ms.
	at Timeout._onTimeout (/home/cleyfaye/.vscode/extensions/visualstudioexptteam.vscodeintellicode-completions-1.0.22/dist/extension.js:2:614697)
	at listOnTimeout (node:internal/timers:569:17)
	at process.processTimers (node:internal/timers:512:7)
[inlineCompletionProvider.provideInlineCompletionItems() 15]  Caught error when requesting inline completion: Error: Operation took longer than 5000ms.
[inlineCompletionProvider.provideInlineCompletionItems() 15] Unexpected exception: Operation took longer than 5000ms.
Error: Operation took longer than 5000ms.
	at Timeout._onTimeout (/home/cleyfaye/.vscode/extensions/visualstudioexptteam.vscodeintellicode-completions-1.0.22/dist/extension.js:2:614697)
	at listOnTimeout (node:internal/timers:569:17)
	at process.processTimers (node:internal/timers:512:7)
```

I tried removing the extension and installing it back, same result. There is no suggestion at all.

**Comment by GollyJer** [comment]
+1

**Comment by hgokturk** [comment]
Same here. Created a probably duplicate issue here: #507

**Comment by CleyFaye** [comment]
This issue can be "temporarily fixed" by enabling the "--no-experimental-fetch" option for node 18.
I can confirm that forcing this in the embedded node does allow the "Intellicode Autocomplete" extension to work fine (no more URL error in the output, and actual completion in some TypeScript files).

Versions are the same as the ones in my post above.

The only issue is, I have no idea how to pass this argument to VSCode. I literally edited the binary and replaced `getOptionValue('--no-experimental-fetch')` with a `1` to test this.

It should be easy to fix however, as a rebuild of the extension with up-to-date tools should fix the actual issue…

**Comment by Rony-dot** [comment]
#### **IntelliCode** is not working
#### Below is **output** for **IntelliCode Completion**
```logs
[inlineCompletionProvider.provideInlineCompletionItems() 410]  Caught error when requesting inline completion: Error: Operation took longer than 5000ms.
[inlineCompletionProvider.provideInlineCompletionItems() 410] Unexpected exception: Operation took longer than 5000ms.
Error: Operation took longer than 5000ms.
	at Timeout._onTimeout (/home/rakibul/.vscode/extensions/visualstudioexptteam.vscodeintellicode-completions-1.0.22/dist/extension.js:2:614697)
	at listOnTimeout (node:internal/timers:569:17)
	at process.processTimers (node:internal/timers:512:7)
```

**Comment by AitomHot** [comment]
+1

Version: 1.85.0 (Universal)
Commit: af28b32d7e553898b2a91af498b1fb666fdebe0c
Date: 2023-12-06T18:18:04.614Z
Electron: 25.9.7
ElectronBuildId: 25551756
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin arm64 23.0.0

`[Completions][4]: [Child process] Unhandled rejection error stack: TypeError: Failed to parse URL from /Users/amurcia/.vscode/extensions/ms-dotnettools.vscodeintellicode-csharp-0.1.26-darwin-arm64/dist/node_modules/@vsintellicode/completions-csharp/blingfire.wasm
    at Object.fetch (node:internal/deps/undici/undici:11413:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)`

**Comment by Einxeld** [comment]
Uninstalling vscode and installing v1.81 worked for me.

**Comment by CleyFaye** [comment]
Obviously reverting to an older version with a compatible node version works, but that's hardly a fix.

**Comment by JasonKroll** [comment]
+1 
Version: 1.85.1 (Universal)
Commit: 0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2
Date: 2023-12-13T09:48:06.308Z
Electron: 25.9.7
ElectronBuildId: 25551756
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Darwin x64 22.6.0

```

Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 1]  Completion was filtered out because Completion line is an empty string
[Completions][4]: [Child process] Unhandled rejection error: Failed to parse URL from /Users/jasonkroll/.vscode/extensions/ms-dotnettools.vscodeintellicode-csharp-0.1.26-darwin-x64/dist/node_modules/@vsintellicode/completions-csharp/tree-sitter.wasm
[Completions][4]: [Child process] Unhandled rejection error stack: TypeError: Failed to parse URL from /Users/jasonkroll/.vscode/extensions/ms-dotnettools.vscodeintellicode-csharp-0.1.26-darwin-x64/dist/node_modules/@vsintellicode/completions-csharp/tree-sitter.wasm
    at Object.fetch (node:internal/deps/undici/undici:11413:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Completions][4]: [Child process] Unhandled rejection error: Failed to parse URL from /Users/jasonkroll/.vscode/extensions/ms-dotnettools.vscodeintellicode-csharp-0.1.26-darwin-x64/dist/node_modules/@vsintellicode/completions-csharp/blingfire.wasm
[Completions][4]: [Child process] Unhandled rejection error stack: TypeError: Failed to parse URL from /Users/jasonkroll/.vscode/extensions/ms-dotnettools.vscodeintellicode-csharp-0.1.26-darwin-x64/dist/node_modules/@vsintellicode/completions-csharp/blingfire.wasm
    at Object.fetch (node:internal/deps/undici/undici:11413:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 2]  Entered provideInlineCompletionItems() try block
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 2] Skipping request
[Completions][2]: [CompletionState.getLineCompletion()] : Best suffix
	First token score log confidence: 0 < -0.35667494393873245
	Suggestion: 
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 2]  Candidate completion: 
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 2]  Current line with candidate:

[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 2]  Completion was filtered out because Completion line is an empty string
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 3]  Entered provideInlineCompletionItems() try block
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 3] Starting request
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 3]  Recieved cancellation event, ran for 461.0476541519165
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 4]  Entered provideInlineCompletionItems() try block
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 4] Skipping request
[Completions][2]: [CompletionState.getLineCompletion()] : Best suffix
	First token score log confidence: 0 < -0.35667494393873245
	Suggestion: 
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 4]  Candidate completion: 
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 4]  Current line with candidate:

[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 4]  Completion was filtered out because Completion line is an empty string
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 5]  Entered provideInlineCompletionItems() try block
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 5] Starting request
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 5]  Recieved cancellation event, ran for 28.407052993774414
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 6]  Entered provideInlineCompletionItems() try block
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 6] Starting request
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 6]  Recieved cancellation event, ran for 451.55600118637085
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 7]  Entered provideInlineCompletionItems() try block
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 7] Starting request
[Completions][4]: [inlineCompletionProvider.provideInlineCompletionItems() 0]  Caught error when requesting inline completion: Error: Operation took longer than 5000ms.
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 0] 
	Wait: 0.675900936126709ms
	Full inference: 5002.836025238037
[Completions][4]: [inlineCompletionProvider.provideInlineCompletionItems() 0] Unexpected exception: Operation took longer than 5000ms.
Error: Operation took longer than 5000ms.
	at Timeout._onTimeout (/Users/jasonkroll/.vscode/extensions/ms-dotnettools.vscodeintellicode-csharp-0.1.26-darwin-x64/dist/node_modules/@vsintellicode/completions-csharp/index.js:2:66913)
	at listOnTimeout (node:internal/timers:569:17)
	at process.processTimers (node:internal/timers:512:7)
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 3] Cancelled after 3178.7894020080566ms wait
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 5] Cancelled after 2437.795147895813ms wait
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 6] Cancelled after 2338.8278398513794ms wait
[Completions][1]: [WorkerLineCompletionService.request() 8]  Entered
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 7]  Recieved cancellation event, ran for 6236.179682731628
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 8]  Entered provideInlineCompletionItems() try block
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 8] Skipping request
[Completions][2]: [CompletionState.getLineCompletion()] : Best suffix
```

**Comment by maciekglowka** [comment]
Still persists on 1.86.0 that I've updated today.

**Comment by Useroth** [comment]
> Still persists on 1.86.0 that I've updated today.

+1

**Comment by joanbarros** [comment]
+1

I've been getting this issue for some time now...

```
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 297]  Entered provideInlineCompletionItems() try block
[Completions][1]: [inlineCompletionProvider.provideInlineCompletionItems() 297] Starting request
[Completions][1]: [WorkerLineCompletionService.request() 298]  Entered
[Completions][4]: [inlineCompletionProvider.provideInlineCompletionItems() 297]  Caught error when requesting inline completion: Error: Operation took longer than 5000ms.
[Completions][2]: [inlineCompletionProvider.provideInlineCompletionItems() 297] 
	Wait: 0.4018999934196472ms
	Full inference: 5007.036899983883
[Completions][4]: [inlineCompletionProvider.provideInlineCompletionItems() 297] Unexpected exception: Operation took longer than 5000ms.
Error: Operation took longer than 5000ms.
	at Timeout._onTimeout (c:\Users\barro\.vscode\extensions\ms-dotnettools.vscodeintellicode-csharp-0.1.26-win32-x64\dist\node_modules\@vsintellicode\completions-csharp\index.js:2:66913)
	at listOnTimeout (node:internal/timers:569:17)
	at process.processTimers (node:internal/timers:512:7)
```

**Comment by erdemkucuk** [comment]
Any news about the fix?

**Comment by AlexandreBossard** [comment]
Still not working with 1.87

**Comment by vivlimmsft** [comment]
We have been working on an update that fixes this. Unfortunately it's been a lot slower to get things moving again after so long. I'll post again here when we've released.

Thanks @CleyFaye for the investigation and identifying the `--no-experimental-fetch` flag as a workaround. That is what we'll be using initially since we have multiple upstream wasm dependencies to track down / create updated builds of.

**Comment by vivlimmsft** [comment]
Sorry about the very long delay on resolving this, but I just published [v2.0.0 to the marketplace](https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.vscodeintellicode-completions) which *should* fix this issue.

If you see any problems with v2.0.0 (or later versions) please open a new issue. Thanks!
